### PR TITLE
Initial crate setup, graphics context and client handlers

### PR DESCRIPTION
**Pull request by carlOSx64**
This program currently does nothing from the user's perspective. It only contains the code that establishes a connection between our software and the hardware. Context is the core `struct` in which resources and handlers will be stored.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/target
```
Comment (nilgoyette) [inline .gitignore:1]: Please add `Cargo.lock` in this list and remove it from git.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "diffusion-slice"
```
Comment (nilgoyette) [inline Cargo.toml]: Plerase name it `diffusion-slice-rs`

File: .gitignore (hunk 1 above)
Comment (carlOSx64) [reply .gitignore:1]: I wish I could, but I read that it's recommended to keep it, especially for applications. Apparently, it keeps build dependencies consistent across environments.

Do you want me to remove it anyway?

Comment (nilgoyette) [reply .gitignore:1]: I understand why a company would have such a requirement, but I don't see the advantage in this project. Please remove it. We will add it back if we realize that it's needed some some reasons.

**Pull request merged**